src/renderers/shaders/UniformsUtils.js

```diff
@@ -36,31 +36,37 @@ THREE.UniformsUtils = {
 
 				var parameter_src = uniforms_src[ u ][ p ];
 
-				if ( parameter_src instanceof THREE.Color ||
-					 parameter_src instanceof THREE.Vector2 ||
-					 parameter_src instanceof THREE.Vector3 ||
-					 parameter_src instanceof THREE.Vector4 ||
-					 parameter_src instanceof THREE.Matrix3 ||
-					 parameter_src instanceof THREE.Matrix4 ||
-					 parameter_src instanceof THREE.Texture ) {
+				uniforms_dst[ u ][ p ] = this.cloneValue( parameter_src );
 
-					uniforms_dst[ u ][ p ] = parameter_src.clone();
+			}
 
-				} else if ( Array.isArray( parameter_src ) ) {
+		}
 
-					uniforms_dst[ u ][ p ] = parameter_src.slice();
+		return uniforms_dst;
 
-				} else {
+	},
 
-					uniforms_dst[ u ][ p ] = parameter_src;
+	cloneValue: function( parameter_src ) {
 
-				}
+		if ( parameter_src instanceof THREE.Color ||
+			 parameter_src instanceof THREE.Vector2 ||
+			 parameter_src instanceof THREE.Vector3 ||
+			 parameter_src instanceof THREE.Vector4 ||
+			 parameter_src instanceof THREE.Matrix3 ||
+			 parameter_src instanceof THREE.Matrix4 ||
+			 parameter_src instanceof THREE.Texture ) {
 
-			}
+			return parameter_src.clone();
 
-		}
+		} else if ( Array.isArray( parameter_src ) ) {
```

Comment:
`Array.isArray( new Float32Array( 1 ) ) === false` (!)

Not sure it was handled correctly, before. Looks a lot cleaner now.